pkg/controllermanager/controller/shoot/shoot_control_delete.go

```diff
@@ -90,6 +90,27 @@ func (c *Controller) runDeleteShootFlow(o *operation.Operation) *gardencorev1alp
 		return gardencorev1alpha1helper.LastError(fmt.Sprintf("Failed to retrieve the Shoot namespace in the Seed cluster (%s)", err.Error()))
 	}
 
+	// Unregister the Shoot as Seed cluster if it was annotated to be a seed and is in the garden namespace
+	if o.Shoot.Info.Namespace == common.GardenNamespace && o.ShootedSeed != nil {
+		if err := botanist.UnregisterAsSeed(); err != nil {
+			return gardencorev1alpha1helper.LastError(fmt.Sprintf("Could not unregister Shoot %q as Seed: %+v", o.Shoot.Info.Name, err))
+		}
+
+		// wait seed to be deleted before to go on with the shoot
+		if err = utilretry.UntilTimeout(context.TODO(), time.Second, 10*time.Second, func(context.Context) (done bool, err error) {
+			_, err = c.k8sGardenClient.Garden().GardenV1beta1().Seeds().Get(o.Shoot.Info.Name, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				return utilretry.Ok()
+			}
+			if err != nil {
+				return utilretry.SevereError(err)
+			}
+			return utilretry.Ok()
```

Comment:
In this case The `Seed` still exists. We should not return `Ok`, I guess?

Reply:
You are right. I changed it to `utilretry.NotOk()`